galasa-ui/src/components/test-runs/SearchCriteriaContent.tsx

```diff
@@ -130,38 +129,53 @@ export default function SearchCriteriaContent({requestorNamesPromise, resultsNam
 
   }, [selectedFilter, query]);
 
+  const updateQueryAndUrl = (newQuery: Map<string, string>) => {
+    // Update the component's query state
+    setQuery(newQuery);
   
+    // Synchronize the browser's URL with the new query
+    const params = new URLSearchParams();
+    newQuery.forEach((value, key) => {
+      params.set(key, value);
+    });
+    router.replace(`${pathname}?${params.toString()}`, { scroll: false });
+  };
 
   const handleSave = (event: FormEvent) => {
     event.preventDefault();
-    const newQuery = new Map(query);
 
-    // Update the query based on the selected filter
+    // Determine the new value for the currently selected filter
+    let valueToSet = '';
     if (selectedFilter.id === 'result') {
-      newQuery.set('result', selectedResults.length > 0 ? selectedResults.join(',') : '');
+      valueToSet = selectedResults.join(',');
     } else if (selectedFilter.id === 'status') {
-      newQuery.set('status', selectedStatuses.length > 0 ? selectedStatuses.join(',') : '');
+      valueToSet = selectedStatuses.join(',');
     } else if (selectedFilter.id === 'tags') {
-      newQuery.set('tags', selectedTags.length > 0 ? selectedTags.join(',') : '');
-    }else {
-      newQuery.set(selectedFilter.id, currentInputValue.trim());
+      valueToSet = selectedTags.join(',');
+    } else {
+      valueToSet = currentInputValue.trim();
     }
 
-    // Clean up empty values from the query
-    newQuery.forEach((value, key) => {
-      if (!value) {
-        newQuery.delete(key);
-      }
-    });
-        
-    setQuery(newQuery);
+    // Get the old value from the query state for comparison
+    const oldValue = query.get(selectedFilter.id) || '';
 
-    // Update the URL with the new query parameters
-    const params = new URLSearchParams();
-    newQuery.forEach((value, key) => {
-      params.set(key, value);
-    });
-    router.replace(`${pathname}?${params.toString()}`, {scroll: false});
+    // Only proceed if the value has actually changed.
+    if (valueToSet === oldValue) {
+      return;
+    }
```

Comment:
Really minor - can we avoid returning in the middle of this function? We can invert the condition